VotingApplication/VotingApplication.Web/Scripts/Services/PollStrategy.js

```diff
@@ -0,0 +1,24 @@
+﻿(function () {
+    var VotingApp = angular.module('VotingApp');
+
+    VotingApp.factory('PollStrategy', ['PollAction', function (PollAction) {
+        var pollStrategy;
+
+        PollAction.getPoll(PollAction.currentPollId(), function (data) {
+            pollStrategy = data.VotingStrategy;
+        });
+
+        var getVotingTemplate = function () {
+            if (!pollStrategy) {
+                return 'routes/BasicVote';
+            }
+
+            return 'routes/' + pollStrategy + 'Vote';
+        }
+
+        return {
+            hasVotingTemplate: function() { return pollStrategy != undefined; },
+            getVotingTemplate: getVotingTemplate
```

Comment:
I feel this should just be called votingTemplate instead of getVotingTemplate. Binding a element to getVotingStrategy seems a bit strange.